src/parser/youtube/LiveChat.ts

```diff
@@ -180,6 +190,31 @@ class LiveChat extends EventEmitter {
     return data.actions.array().as(AddChatItemAction);
   }
 
+  async getItemMenu(item: ChatItemHasMenuEndpoint): Promise<Menu> {
```

Comment:
Anyway we can have this method return a new object? Here's a small gist to demonstrate what I mean:

> ./youtube/ItemMenu.ts
```ts
//...

class ItemMenu {
  constructor(data: ParsedResponse) {
    this.#page = data;
  }
  
  /**
   * This should find and select the correct menu item.
   * So "title" would be either the label, text or icon_type.
   * If an item is not matched then throw an error. If it is matched
   * then call its endpoint.
   */
  async selectItem(title: string): Promise<ActionsResponse> {
    // ...
  }
  
  /**
   * Returns original InnerTube response.
   */
  async page(): ParsedResponse {
    return this.#page;
  }
}
```

Then `getItemMenu` in `LiveChat` at line 193 would return an `ItemMenu` object, which should take care of selecting items and calling their endpoints. Ex:

> ./youtube/LiveChat.ts
```js
//...
const response = await item.menu_endpoint.call(this.#actions, undefined, true);
return new ItemMenu(response);
```

This would simplify usage quite a lot, let me know what you think. Thanks!

Reply:
Thank you for the review. I have some worries.

1. When `ItemMenu.selectItem()` is called, to request to YouTube, we need to access LiveChat#actions. However, because `LiveChat#actions` is private member, we cannot acces it inside from `ItemMenu`. So some way to access is needed.
My idea is that LiveChat should have a method to call request to YouTube like call():
```typescript
class LiveChat {
  public async call(endpoint: NavigationEndpoint) {
    const response = await endpoint.call(this.#actions, undefined, true); 
    return response;
  }
}
```
and when `ItemMenu.selectItem()` is called, LiveChat.call() will be called inside from `ItemMenu.selectItem()`.

2. `LiveChatTextMessage.inline_action_buttons` and `LiveChatAutoModMessage.moderation_buttons` should be typed as ItemMenu, right?

3. Selecting item by its label, text or icon_type is simple especially deleting chat, but this may cause unstable works. `label` and `text` changes if another language is selected when `InnerTube.create()` is called. `icon_type` does not changes, but in some case, it is difficult to predict action from icon_type. For example, the icon_type for `hide this user` is `REMOVE_CIRCLE`.
So, I want additional way to call endpoint: receive endpoint instanses (currently typed `Button`) and select the endpoint by endpoint itself. I mean `LiveChat.selectButtonItem(button: Button)` is still needed.

Reply:
> When ItemMenu.selectItem() is called, to request to YouTube, we need to access LiveChat#actions. However, because LiveChat#actions is private member, we cannot acces it inside from ItemMenu. So some way to access is needed.
My idea is that LiveChat should have a method to call request to YouTube like call():

You don't need to create a new method ~
I think simply sharing the current `Actions` instance from `LiveChat` with the `ItemMenu` class would do, ex:
```js
class ItemMenu {
  #page;
  #actions;

  constructor(data: ParsedResponse, actions: Actions) {
    this.#page = data;
    this.#actions = actions;
  }
  // ...
}
```

Then:
```js
//...
const response = await item.menu_endpoint.call(this.#actions, undefined, true);
return new ItemMenu(response, this.#actions);
```
----

> LiveChatTextMessage.inline_action_buttons and LiveChatAutoModMessage.moderation_buttons should be typed as ItemMenu, right?

Nope, since `ItemMenu` will be a wrapper around actual menus it shouldn't interfere with individual parser types.

----

>  So, I want additional way to call endpoint: receive endpoint instanses (currently typed `Button`) and select the endpoint by endpoint itself. I mean `LiveChat.selectButtonItem(button: Button)` is still needed.

Yes, I agree. For that, you can simply expose items from the menu as public properties in the `ItemMenu` class and add an additional method or make `ItemMenu#selectItem` accept either a string or an item of type `Button`.

Reply:
OK! Thank you for your advice! Now I understand your idea.
I will work on to follow your design and request a review again.